org.ekstep.questionunit.mcq-1.2/renderer/plugin.js

```diff
@@ -80,19 +81,41 @@ org.ekstep.questionunitmcq.RendererPlugin = org.ekstep.contentrenderer.questionU
       }
     });
     if (!_.isUndefined(MCQController.pluginInstance._selectedIndex)) telValues['option' + MCQController.pluginInstance._selectedIndex] = selectedAnsData.image.length > 0 ? selectedAnsData.image : selectedAnsData.text; // eslint-disable-line no-undef
+    var params = instance.getTelemetryParams();
     result = {
       eval: correctAnswer,
       state: {
         val: MCQController.pluginInstance._selectedIndex, // eslint-disable-line no-undef
         options: option // eslint-disable-line no-undef
       },
       score: correctAnswer ? MCQController.pluginInstance._question.config.max_score : 0, // eslint-disable-line no-undef
-      values: [telValues]
-    }
+      params: params,
+      resValues: instance.getTelemetryResValues(),
+    };
     if (_.isFunction(callback)) {
       callback(result);
     }
   },
+  getTelemetryParams: function() {
+    // Any change in the index value affects resvalues as well
+    var instance = this;
+    var params = [], qTitle = {}, questionData = MCQController.pluginInstance._question.data;
+    qTitle.title = instance.getTelemetryParamsValue(questionData.question);
+    params.push(qTitle);
+    questionData.options.forEach(function (option,key) { // eslint-disable-line no-undef
+      var temp = {};
+      temp[key+1] = instance.getTelemetryParamsValue(option);
+      params.push(temp);
+    });
+    return params;
+  },
+  getTelemetryResValues: function() {
+    var resValues = [];
+    var selectedIndex = MCQController.pluginInstance._selectedIndex;
+    if (!_.isUndefined(MCQController.pluginInstance._selectedIndex)) 
+    resValues[selectedIndex + 1 ] = this.getTelemetryParamsValue(MCQController.pluginInstance._question.data.options[selectedIndex]);
```

Comment:
This should be array of objects.
example: resValues = [ { "2": "{'text' : 'option text'}"}]